### PR TITLE
Increase Coverage

### LINKED ISSUE #185: Increase coverage by 2.5%

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Added some tests to support the wallet message interface, and in the process found some ways to shore up some bloated code.

## Links to any relevant issues

- Closes #185

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

File: sdk/tests/client/input_selection/alias_outputs.rs [hunk 1]
```diff
@@ -326,6 +338,79 @@ fn burn_alias() {
     assert!(unsorted_eq(&selected.outputs, &outputs));
 }
 
+#[test]
+fn alias_builder_replace() {
```
Comment (thibault-martinez) [inline sdk/tests/client/input_selection/alias_outputs.rs]: Would be nice to move these to `tests/types/*_output.rs`, input_selection is just for input selection

Comment (DaughterOfMars) [reply sdk/tests/client/input_selection/alias_outputs.rs]: Good point.

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/types/block/output/nft.rs [hunk 2]
```diff
@@ -618,3 +618,137 @@ pub mod dto {
         }
     }
 }
+
+#[cfg(test)]
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/nft.rs:622]: Am I too annoying if I ask you to move these to the `tests` folder for consistency? Or do they need private access or something?

File: bindings/core/src/method_handler/call_method.rs [hunk 3]
```diff
@@ -17,6 +20,37 @@ use crate::{
     UtilsMethod,
 };
 
+pub trait CallMethod {
```
Comment (thibault-martinez) [inline bindings/core/src/method_handler/call_method.rs:23]: I wonder if this is actually worth it for something that is not really user facing

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Rebase to include the CI fix

File: sdk/src/types/block/output/nft.rs (hunk 2 above)
Comment (DaughterOfMars) [reply sdk/src/types/block/output/nft.rs:622]: The dto one does need private access.

File: bindings/core/src/method_handler/call_method.rs (hunk 3 above)
Comment (DaughterOfMars) [reply bindings/core/src/method_handler/call_method.rs:23]: I found the public functions that are currently used to be very...anti-idomatic.

File: sdk/src/types/block/output/nft.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/types/block/output/nft.rs:622]: Fair enough

**Review by Thoralf-M — APPROVED** [review]

File: bindings/core/src/method_handler/call_method.rs (hunk 3 above)
Comment (thibault-martinez) [reply bindings/core/src/method_handler/call_method.rs:23]: Yeah I agree, was just weighing pros and cons of convenience vs an additional Box

Comment (thibault-martinez) [reply bindings/core/src/method_handler/call_method.rs:23]: Cause in the end these functions are reaching an extremely tiny set of people actually writing bindings, which is mostly... us

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**